src/msw/notebook.cpp

```diff
@@ -729,22 +730,195 @@ bool wxNotebook::InsertPage(size_t nPage,
+    if (len)
+    {
+        // This is intentional and not a bug. :) On left/right the font is rotated 90 degree
+        // so we have to account for that by flipping width with height.
+        GetTextExtent(label, &tr.height, &tr.width, nullptr, nullptr, &labelFont);
+    }
+
+    // The tab has neither a text nor an icon
+    if (len == 0 && ir.width == 0)
+    {
+        *flags = wxBK_HITTEST_ONITEM;
+        return true;
+    }
+
+    if (len)
+    {
+        tr.x = r.x + (r.width - tr.width)/2;
+        if (tabDir == wxLEFT)
+        {
+            if (ir.height)
+                tr.y = r.y + (r.height - tr.height - gap - ir.height) / 2;
+            else
+                tr.y = r.y + (r.height - tr.height - ir.height) / 2;
+        }
+        else
+        {
+            if (ir.height)
+                tr.y = r.y + (r.height - tr.height + gap + ir.height) / 2;
+            else
+                tr.y = r.y + (r.height - tr.height) / 2;
+        }
+
+        if (tr.Contains(pt))
+        {
+            *flags = wxBK_HITTEST_ONLABEL;
+            return true;
+        }
+    }
+
+    if (ir.height)
+    {
+        wxDisplay disp(wxDisplay::GetFromWindow(this));
+        double scale = disp.GetScaleFactor();
+        int dispIndex = 0;
+        // 1.0 = 3
+        // 1.25 = 8
+        // 1.50 = 14
+        // 1.75 = 3
+        // 2.00 = 6
+        // 2.25 = 12
+        if (scale > 1.0f)
+        {
+            if (scale <= 1.25f)
+                dispIndex = 1;
+            else if (scale <= 1.5f)
+                dispIndex = 2;
+            else if (scale <= 1.75f)
+                dispIndex = 0;
+            else if (scale <= 2.0f)
+                dispIndex = 3;
+            else
+                dispIndex = 4;
+        }
+
+        static int displacementTab[5] = { 3, 8, 14, 6, 12 };
+        int displacement = displacementTab[dispIndex];
+
+        ir.x = r.x + (r.width - ir.width) / 2;
+        if (tabDir == wxLEFT)
+        {
+            if (len)
+                ir.y = tr.y + tr.height + gap;
+            else
+                ir.y = r.y - displacement + (r.height - ir.height)/2;
+        }
+        else
+        {
+            if (len)
+                ir.y = r.y + (r.height - ir.height -gap -tr.height) / 2;
+            else
+                ir.y = r.y - displacement + (r.height - ir.height) / 2;
+        }
+
+        if (ir.Contains(pt))
+        {
+            *flags = wxBK_HITTEST_ONICON;
+            return true;
+        }
+    }
+
+    *flags = wxBK_HITTEST_ONITEM;
+    return true;
+}
+
 int wxNotebook::HitTest(const wxPoint& pt, long *flags) const
 {
+    int item = -1;
+
     TC_HITTESTINFO hitTestInfo;
     hitTestInfo.pt.x = pt.x;
     hitTestInfo.pt.y = pt.y;
-    int item = TabCtrl_HitTest(GetHwnd(), &hitTestInfo);
+    item = TabCtrl_HitTest(GetHwnd(), &hitTestInfo);
 
-    if ( flags )
+    if (flags)
     {
         *flags = 0;
 
+        if (MSWHitTestLeftRight(pt, item, flags))
+            return item;
```

Comment:
Sorry, but this still doesn't seem right: now we only call this function if `flags != 0`. But if I understand your intention, we should always call it -- and check if `flags` is valid inside it.